File: docs/coding/samples/numcal.py
def main():
    print("=== Number Calculator ===")
    
    numbers = []
    
    print("Enter numbers (type 'done' to finish):")
    
    while True:
        user_input = input("> ").strip()
        
        if user_input.lower() == 'done':
            break
        
        try:
            num = float(user_input)
            numbers.append(num)
        except ValueError:
            print(f"'{user_input}' is not a valid number. Try again.")
    
    if not numbers:
        print("No numbers entered.")
        return
    
    total = sum(numbers)
    average = total / len(numbers)
    maximum = max(numbers)
    minimum = min(numbers)
    
    print(f"\nResults:")
    print(f"Count: {len(numbers)}")
    print(f"Sum: {total:.2f}")
    print(f"Average: {average:.2f}")
    print(f"Max: {maximum:.2f}")
    print(f"Min: {minimum:.2f}")
    
    print(f"\nNumbers entered: {numbers}")

if __name__ == "__main__":
    main()

File: docs/coding/samples/numcal.java
import java.util.ArrayList;
import java.util.Scanner;

public class NumberCalculator {
    public static void main(String[] args) {
        System.out.println("=== Number Calculator ===");
        
        Scanner scanner = new Scanner(System.in);
        ArrayList<Double> numbers = new ArrayList<>();
        
        System.out.println("Enter numbers (type 'done' to finish):");
        
        while (true) {
            String input = scanner.nextLine().trim();
            
            if (input.equalsIgnoreCase("done")) {
                break;
            }
            
            try {
                double num = Double.parseDouble(input);
                numbers.add(num);
            } catch (NumberFormatException e) {
                System.out.printf("'%s' is not a valid number. Try again.%n", input);
            }
        }
        
        scanner.close();
        
        if (numbers.isEmpty()) {
            System.out.println("No numbers entered.");
            return;
        }
        
        double sum = 0;
        double max = Double.NEGATIVE_INFINITY;
        double min = Double.POSITIVE_INFINITY;
        
        for (double num : numbers) {
            sum += num;
            if (num > max) max = num;
            if (num < min) min = num;
        }
        
        double average = sum / numbers.size();
        
        System.out.printf("%nResults:%n");
        System.out.printf("Count: %d%n", numbers.size());
        System.out.printf("Sum: %.2f%n", sum);
        System.out.printf("Average: %.2f%n", average);
        System.out.printf("Max: %.2f%n", max);
        System.out.printf("Min: %.2f%n", min);
        
        System.out.printf("%nNumbers entered: %s%n", numbers);
    }
}